src/pudl/metadata/resources/allocate_net_gen.py

```diff
@@ -0,0 +1,139 @@
+                    "prime_mover_code",
+                    "energy_source_code",
+                    "energy_source_code_num",
+                    "net_generation_mwh",
+                    "fuel_consumed_mmbtu",
+                    "fuel_consumed_for_electricity_mmbtu",
+                ],
+                "primary_key": [
+                    "report_date",
+                    "plant_id_eia",
+                    "generator_id",
+                    "prime_mover_code",
+                    "energy_source_code",
+                ],
+            },
+            "field_namespace": "eia",
+            "sources": ["eia923", "eia860"],
+            "etl_group": "outputs",
+        }
+        for freq in AGG_FREQS
+    }
+    | {
+        f"generation_fuel_by_generator_{freq}_eia923": {
+            "description": (
+                f"{freq.title()} estimated net generation by generator. Based on net "
+                "generation reported in the EIA-923 generation and generation_fuel tables."
+                "The net electricity generation and fuel consumption reported in the EIA-923 "
+                "generation fuel are allocated to individual generators. This process does not "
+                "distinguish between primary and secondary energy_sources for generators. Net "
+                "generation is allocated equally between energy source codes, so if a "
+                "plant has multiple generators with the same prime_mover_code but different "
+                "energy source codes the generation_fuel_eia923 records will be associated "
+                "similarly between these two generators. Allocated net generation will still "
+                "be proportional to each generator's net generation or capacity."
+            ),
+            "schema": {
+                "fields": [
+                    "report_date",
+                    "plant_id_eia",
+                    "plant_id_pudl",
+                    "plant_name_eia",
+                    "utility_id_eia",
+                    "utility_id_pudl",
+                    "utility_name_eia",
+                    "generator_id",
+                    "unit_id_pudl",
+                    "fuel_consumed_for_electricity_mmbtu",
+                    "fuel_consumed_mmbtu",
+                    "net_generation_mwh",
+                ],
+                "primary_key": [
+                    "report_date",
+                    "plant_id_eia",
+                    "generator_id",
+                ],
+            },
+            "field_namespace": "eia",
+            "sources": ["eia923", "eia860"],
+            "etl_group": "outputs",
+        }
+        for freq in AGG_FREQS
+    }
+    | {
+        "generation_fuel_by_generator_energy_source_owner_yearly_eia923": {
+            "description": (
+                "Yearly estimated net generation and fuel consumption for each generator, "
+                "broken down by energy source, prime mover, and owner. Based on data "
```

Comment:
It's unclear from this description whether we are aggregating to individual prime movers, or within all prime movers of a given type within a plant.

It might make sense to pull out the text which is shared between all of these longer generation fuel descriptions into a module-level constant so that it can be re-used and edited without needing to be changed in several places.